vllm_detector_adapter/generative_detectors/granite_guardian.py

```diff
@@ -95,6 +111,124 @@ def __preprocess(
+                        name=tool_call_function["name"], arguments=arg_dict
+                    )
+                    assistant_tool_call_functions.append(new_tool_call_function)
+                assistant_content_string = orjson.dumps(
+                    assistant_tool_call_functions, option=self.INDENT
+                ).decode()
+                assistant_message = DetectionChatMessageParam(
+                    role=message["role"], content=assistant_content_string
+                )
+                assistant_message_count += 1
+
+            # (2) User messages are just passed through
+            if message["role"] == "user":
+                user_message = message
+                user_message_count += 1
+
+        # Error if no assistant message was found
+        if not assistant_message or not assistant_message["content"]:
+            return ErrorResponse(
+                message="no assistant message was provided with tool_calls for analysis",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+        # Error if no user message was found
+        if not user_message or not user_message["content"]:
+            return ErrorResponse(
+                message="no user message was provided with content for analysis",
+                type="BadRequestError",
+                code=HTTPStatus.BAD_REQUEST.value,
+            )
+        # Warning if multiple assistant messages were found
+        if assistant_message_count > 1:
+            logger.warning(
+                "More than one assistant message with tool_calls was provided. Only the last will be used for analysis with tools."
+            )
+        # Warning if multiple user messages were found
+        if user_message_count > 1:
+            logger.warning(
+                "More than one user message was provided. Only the last will be used for analysis with tools."
+            )
+
+        # 'tools' ref https://platform.openai.com/docs/api-reference/chat/create#chat-create-tools
+        # are provided on the same level as 'messages' on the user request to this
+        # adapter. 'tools' are generally a list of tools the model may call. In the Granite
+        # Guardian context, these are the definition of tools to be used for analysis.
+        # (3) Provide the inner tools functions as a message with role: tools
+        tools_functions = []
+        for tools in request.tools:
+            tools_functions.append(tools["function"])
+        tools_content_string = orjson.dumps(
+            tools_functions, option=self.INDENT
+        ).decode()
+        tools_message = DetectionChatMessageParam(
+            role="tools", content=tools_content_string
+        )
+
+        # Replace request portions
+        # `tools` should not be passed onto the chat completions request currently, but we
+        # do not pass them on here anyway, in case the chat completion request processing
```

Comment:
```suggestion
        # do pass them on here anyway, in case the chat completion request processing
```

?

Reply:
Ah I see.. I thought the sentence was trying to negate previous one.. reading the code clarified.